### PR TITLE
ansible: use explicit tap2junit version

### PR DESCRIPTION
**Pull request by MoLow**
Fixes: https://github.com/nodejs/build/issues/3268
Fixes: https://github.com/nodejs/node/issues/47048

### CODE REVIEW

**Comment by MoLow** [comment]
I am using `--step`

**Review by richardlau — APPROVED** [review]

**Comment by richardlau** [comment]
> @richardlau I have tested this using `ansible-playbook ansible/playbooks/jenkins/worker/create.yml --limit "test-rackspace-debian10-x64-1" --start-at-task "prepare installing tap2junit" -vv`
> 
> is there a good way to stop after this task?

I'm not aware of one.

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by MoLow** [comment]
upgraded: 
- windows
- aix, aix73
- freebsd
- macos, macos10
- smartos
- ubuntu

File: ansible/roles/jenkins-worker/tasks/partials/tap2junit/ubuntu.yml [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
-# ubuntu series: python 2.7
+# ubuntu series
 #
 
 - name: install pip
-  package: name=python-pip state=present
+  when: os != "ubuntu1604"
+  package: name=python3-pip state=present
+
+# RHEL 7 doesn't have a package for pip and EPEL 7 doesn't support s390x.
```
Comment (richardlau) [inline ansible/roles/jenkins-worker/tasks/partials/tap2junit/ubuntu.yml:11]: This file (`ubuntu.yml`) won't be run on the RHEL 7 machines.

File: ansible/roles/jenkins-worker/tasks/partials/tap2junit/ubuntu.yml [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+# Install pip via `get-pip.py`.
+- name: ubuntu1604 | download pip install script
+  when: os == "ubuntu1604"
```
Comment (richardlau) [inline ansible/roles/jenkins-worker/tasks/partials/tap2junit/ubuntu.yml:14]: I'm not sure any of the ubuntu1604 machines are connected to the test CI as they are still running Java 8. I'd prefer to orderly remove them (first from Jenkins, then Ansible) and then replace them with something newer (e.g. Ubuntu 20.04 or 22.04).

**Comment by richardlau** [comment]
@MoLow I'm wondering if the Python pip error in https://ci.nodejs.org/job/node-test-linter/49235/console is related to the changes in this PR. It ran on [test-ibm-ubuntu1804-x64-1](https://ci.nodejs.org/computer/test-ibm-ubuntu1804-x64-1).

<details>

```console
00:25:23 + make lint-py-build PYTHON=python3
00:25:24 Pip installing flake8 linter on Python 3.8.0...
00:25:24 python3 -m pip install --no-user --upgrade -t tools/pip/site-packages flake8 || \
00:25:24 	python3 -m pip install --no-user --upgrade --system -t tools/pip/site-packages flake8
00:25:26 
00:25:26 Usage:   
00:25:26   /usr/bin/python3 -m pip install [options] <requirement specifier> [package-index-options] ...
00:25:26   /usr/bin/python3 -m pip install [options] -r <requirements file> [package-index-options] ...
00:25:26   /usr/bin/python3 -m pip install [options] [-e] <vcs project url> ...
00:25:26   /usr/bin/python3 -m pip install [options] [-e] <local project path> ...
00:25:26   /usr/bin/python3 -m pip install [options] <archive url/path> ...
00:25:26 
00:25:26 no such option: --no-user
00:25:27 
00:25:27 Usage:   
00:25:27   /usr/bin/python3 -m pip install [options] <requirement specifier> [package-index-options] ...
00:25:27   /usr/bin/python3 -m pip install [options] -r <requirements file> [package-index-options] ...
00:25:27   /usr/bin/python3 -m pip install [options] [-e] <vcs project url> ...
00:25:27   /usr/bin/python3 -m pip install [options] [-e] <local project path> ...
00:25:27   /usr/bin/python3 -m pip install [options] <archive url/path> ...
00:25:27 
00:25:27 no such option: --no-user
00:25:27 Makefile:1520: recipe for target 'lint-py-build' failed
```
</details>

**Comment by MoLow** [comment]
@richardlau I changed the ubuntu script and the linter is now fixed
however I had some trouble updating the python version on aix

**Comment by cclauss** [comment]
Where is the `--no-user` option that caused #3273 ?

Why is #3273 closed if this PR that fixes it is still open?

A `--no-user` option was added to pip 5 years ago but was later removed.
* https://github.com/pypa/pip/pull/5116/files

File: ansible/roles/jenkins-worker/tasks/partials/tap2junit/ibmi.yml [hunk 3]
```diff
@@ -6,4 +6,4 @@
 
 - name: install tap2junit
   shell:
```
Comment (cclauss) [inline ansible/roles/jenkins-worker/tasks/partials/tap2junit/ibmi.yml:8]: ⛔ At end of file below.
Screenshot 2023-03-31 at 07 43 45

File: ansible/roles/jenkins-worker/vars/main.yml [hunk 4]
```diff
@@ -133,3 +133,5 @@ jobs_variants: {
   arm64: '3',
   smartos: '2'
 }
```
Comment (cclauss) [inline ansible/roles/jenkins-worker/vars/main.yml:135]: ⛔ At end of file below.

**Review by cclauss — APPROVED** [review]

**Comment by MoLow** [comment]
> Where is the --no-user option that caused https://github.com/nodejs/build/issues/3273 ?

https://github.com/nodejs/node/blob/main/Makefile#L1521-L1522

> Why is https://github.com/nodejs/build/issues/3273 closed if this PR that fixes it is still open?

the fix was deployed manually

**Comment by richardlau** [comment]
> @richardlau I changed the ubuntu script and the linter is now fixed (https://ci.nodejs.org/job/node-test-linter/49246/) however I had some trouble updating the python version on aix

Did you sort out the issue on AIX? If not, what are you seeing?

**Comment by MoLow** [comment]
> Did you sort out the issue on AIX? If not, what are you seeing?

I am not able to install python > 3.6, see https://ci.nodejs.org/job/node-test-commit-aix/
it seems that AIX is the only platform not yet using the latest version of tap2junit and causing the build to not bee "green"

**Pull request merged**